front-end/src/AdminBar/AdminView.js

```diff
@@ -1,91 +1,84 @@
 import React from "react";
 import { addOrRemove, clear } from "../utils/api";
-import { AdminPanel, FocusInput, ColoredButton } from "./adminStyles";
-import {
-  ESCAPE_KEY,
-  ENTER_KEY,
-  NUMPAD_ENTER_KEY,
-  BACKSPACE,
-  DELETE
-} from "../utils/keypadNumbers";
+import { AdminForm, FocusInput, ColoredButton } from "./adminStyles";
+
+const MAX_NUMBER_LENGTH = 8;
+const filterNumeric = str => str.replace(/\D/g,'');
 
 class AdminView extends React.Component {
   constructor(props) {
     super(props);
     this.state = { typed: "" };
+    this.inputRef = React.createRef();
+    this.onFormSubmit = this.onFormSubmit.bind(this);
   }
 
-  // Implemented this way to always accept input!
-  handleKeyDown = event => {
-    const keyCode = event.keyCode;
-    switch (keyCode) {
-      case ESCAPE_KEY:
-        this.setState({ typed: "" });
-        break;
-      case ENTER_KEY:
-      case NUMPAD_ENTER_KEY:
-        this.sendNumber();
-        break;
-      case DELETE:
-      case BACKSPACE:
-        this.setState({
-          typed: this.state.typed.substring(0, this.state.typed.length - 1)
-        });
-        break;
-      default:
-        this.handleIfNumber(keyCode);
-    }
-  };
-
   sendNumber = () => {
     this.sendNumberWithParam(this.state.typed);
   };
 
   sendNumberWithParam = num => {
     if (num.length > 0) {
       addOrRemove(num);
-      this.setState({
-        typed: ""
-      });
+      this.setState({ typed: "" });
     }
   };
 
-  undo = () => {
-    // Not yet implemented use redux
-  };
-
-  handleIfNumber(keyCode) {
-    if (this.state.typed.length < 8) {
-      // max len accpt by backend
-      const inputSymbol = String.fromCharCode(keyCode);
-      if (inputSymbol && !isNaN(inputSymbol)) {
-        this.setState({ typed: this.state.typed.concat(inputSymbol) });
-      }
+  clearOnEscape = (event) => {
+    if (event.key === "Escape") {
+      this.setState({ typed: "" });
     }
   }
 
-  componentWillMount() {
-    document.addEventListener("keydown", this.handleKeyDown.bind(this));
+  onFormSubmit(event) {
+    event.preventDefault();
+    this.sendNumber();
+  }
+
+  focusInput = () => {
+    // setTimeout hack necessary to prevent browsers blocking this behaviour
```

Comment:
Haha, nice! Is this something that could get "fixed" by browsers in the future if it is unwanted behaviour?

Reply:
Hmm, I'm not sure tbh, but I doubt it. My guess is that this is either 
1. intentional to block a dark UX pattern, in which case I doubt they'll change their mind (my experience is that browsers protect the users more and more, not the other way around :)), or 
2. that it's simply due to the order in which eventListeners are executed in browsers, in which case it's probably a scenario decided upon in some w3c/tc39 spec and will definitely not change :smile:

Reply:
Hmm, yea it might just be something about ordering. Using set timeout would push it to be excecuted after all the other events have already happened.